content/chapitres/project-lifecycle.adoc

```diff
@@ -1,4 +1,779 @@
+** Simplification des dépendances (pas de version à gérer, grâce au "POM parent")
+** Ajout d'une dépendance
+** Activation du plugin Spring Boot lors des phases de "build"
+
+== Exemple Maven : pom.xml final
+
+[source,xml,subs="+attributes"]
+----
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+    <groupId>com.cicdlectures</groupId>
+    <artifactId>menuserver</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <parent>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-parent</artifactId>
+      <version>{springboot_version}</version>
+    </parent>
+
+    <properties>
+      <java.version>11</java.version>
+    </properties>
+
+    <dependencies>
+      <dependency>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-data-jpa</artifactId>
+      </dependency>
+      <dependency>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-web</artifactId>
+      </dependency>
+      <dependency>
+			<groupId>org.projectlombok</groupId>
+        <artifactId>lombok</artifactId>
+      </dependency>
+      <dependency>
+        <groupId>com.h2database</groupId>
+        <artifactId>h2</artifactId>
+        <scope>runtime</scope>
+      </dependency>
+	</dependencies>
+
+  <build>
+    <plugins>
+        <plugin>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-maven-plugin</artifactId>
+        </plugin>
+    </plugins>
+  </build>
+</project>
+----
+
+== Exercice : Démarrer l'application
+
+* *But*: Exécuter l'application à l'aide du plugin Spring Boot
+
+* Sprint Boot fournit une phase Maven nommée `spring-boot:run` qui exécute
+l'application en mode "développement" sur le port `8080` local
+** Essayez d'appeler cette phase avec Maven
+
+* Résultat attendu (une jolie bannière ASCIIArt):
++
+[source,subs="+attributes"]
+----
+
+  .   ____          _            __ _ _
+ /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
+( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
+ \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
+  '  |____| .__|_| |_|_| |_\__, | / / / /
+ =========|_|==============|___/=/_/_/_/
+ :: Spring Boot ::                (v{springboot_version})
+----
+
+== Exercice : Prévisualiser l'application
+
+* Dans un second terminal de Gitpod, affichez la page web de l'application avec les commandes suivantes :
+** `gp url 8080` pour afficher l'URL publique de l'application correspondant au port `8080` local
+** `gp preview "$(gp url 8080)/"` pour prévisualiser le "endpoint" `/` dans un navigateur local
+
+* La page d'accueil doit afficher HTTP/404
+
+* Trouvez la page des menus qui doit répondre `[]` (liste vide en JSON)
+** 💡 `src/\*/controller/*.java`
```

Comment:
```suggestion
Dans un second terminal de Gitpod, essayez de lister les menus enregistrés dans le menu-server comme présenté dans le chapitre précédent using curl.
```

Nit: We're not really talking about page, but API endpoint?
Also, they already know about this `GET /menus` endpoint from the previous chapter, as we already make them play with curl. 
Soooo, I would say we don't really need to mention gitpod public URLs, only using curl simplifies the whole thing?

WDYT?

Reply:
I'm not at ease with letting them only use `curl`. Being able to visualize through a webbrowser makes it more real.
We can still recommend `curl -v http://localhost:8080` next to it though WDYT?

Reply:
If we do it here, we need to do it also in the previous part then, would be less confusing for the students? (like why we need to expose now, and not before).

Reply:
yep, make sense. Let's add a new PR for this after. Sounds good for you?